app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.fxa
+
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.ProcessLifecycleOwner
+import io.reactivex.subjects.BehaviorSubject
+import kotlinx.coroutines.Deferred
+import mozilla.appservices.fxaclient.Config
+import mozilla.components.concept.sync.AccountObserver
+import mozilla.components.concept.sync.DeviceCapability
+import mozilla.components.concept.sync.DeviceEvent
+import mozilla.components.concept.sync.DeviceEventsObserver
+import mozilla.components.concept.sync.DeviceType
+import mozilla.components.concept.sync.OAuthAccount
+import mozilla.components.concept.sync.Profile
+import mozilla.components.service.fxa.manager.DeviceTuple
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.AUTHENTICATED
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.AUTHENTICATED_NO_PROFILE
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.NEEDS_REAUTHENTICATION
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.NOT_AUTHENTICATED
+
+private const val LOGTAG = "FxaRepo"
+
+/**
+ * Manages Firefox Account (FxA) state. In particular, [accountStateDontUseMeYet] exposes the current sign in
+ * state of the user. If you want to initiate sign in, see [FxaLoginUseCase.beginLogin].
+ *
+ * Devs should use this class rather than interacting with the FxA library directly.
+ */
+class FxaRepo(
+    context: Context,
+    val accountManager: FxaAccountManager = newInstanceDefaultAccountManager(context)
+) {
+
+    enum class AccountState {
+        // TODO: Are these states accurate?
+        // TODO: Later, may need "failed to login": https://github.com/mozilla-mobile/android-components/issues/3712
+        AUTHENTICATED,
+        AUTHENTICATED_NO_PROFILE, // todo: necessary? profile seems to be fetched async so yes?
+        NEEDS_REAUTHENTICATION,
+        NOT_AUTHENTICATED
```

Comment:
Is there any potential overlap between these states?  `NEEDS_REAUTHENTICATION` seems like it might be a substate of `NOT_AUTHENTICATED`, for example.